addons/product/views/product_pricelist_views.xml

```diff
@@ -253,6 +253,7 @@
                                       <field name="date_end"/>
                                       <field name="base" invisible="1"/>
                                       <field name="applied_on" invisible="1"/>
+                                      <field name="company_id" invisible="1"/>
```

Comment:
The product_id and product_template_id should have updated domains taking into account `parent.company_id` because in an inline view, the values aren't properly propagated into the lines (not until the pricelist is saved).

`domain="['|', ('company_id', '=', False), ('company_id', '=', parent.company_id)]`